xdsl/ir/core.py

```diff
@@ -1040,6 +1040,17 @@ def get_attr_or_prop(self, name: str) -> Attribute | None:
             return self.attributes[name]
         return None
 
+    def is_before_in_block(self, other_op: Operation) -> bool:
+        parent_block = self.parent_block()
+        assert isinstance(parent_block, Block)
+
+        if parent_block.get_operation_index(self) < parent_block.get_operation_index(
+            other_op
+        ):
+            return True
+        else:
+            return False
+
```

Comment:
Let's first merge this change in a separate PR? In MLIR, operation indices are cached, which makes this a constant-time comparison if they're cached, and O(length of block) otherwise. In xDSL, getting the operation index is always O(length of block). I would recommend an implementation like this:
```suggestion
    def is_before_in_block(self, other_op: Operation) -> bool:
        if (parent_block := self.parent_block()) is None or other_op.parent_block() is not parent_block:
            return False

        op = self
        while op is not None:
            if op is other_op:
                return True
            op = op.next_op
        return False
```

Reply:
This way the cost is either the distance between the ops if one is before the other, or the length of the tail of the block.

Reply:
PR #4385